src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -200,17 +217,14 @@ protected void install() throws InterruptedException {
 
             if (whatHappened == WhatHappened.initialized || node == null) {
                 updateDeviceGroups(whatHappened, deviceGroupTopics);
-                updateCAType(caTypeTopic);
             } else if (node.childOf(DEVICE_GROUPS_TOPICS)) {
                 updateDeviceGroups(whatHappened, deviceGroupTopics);
-            } else if (node.childOf(CA_TYPE_TOPIC)) {
-                if (caTypeTopic.getOnce() == null) {
-                    return;
-                }
-                updateCAType(caTypeTopic);
+            } else if (node.childOf(CERTIFICATE_AUTHORITY_TOPIC)) {
+                logger.atInfo("service-config-change").kv("event", whatHappened).kv("node", node)
+                        .log("Requesting re-installation of client device auth component");
+                requestReinstall();
```

Comment:
reinstall will force any dependencies of CDA to stop. Make sure this is what we intend.

Reply:
I think that's what we want too because the dependency component is re-installed as well.

Reply:
Dependencies will not be reinstalled.

And why is that what we want? We do not want the MQTT broker to be taken down unnecessarily.

Reply:
We reinstall CDA only when CA related config changes. So, any component that has hard dependency on CDA should be restarted to use the new CA.

Reply:
Why should they _restart_ though? The CA is sent in a subscription based API so that dependencies don't need to restart.

Reply:
Oh right, that works too. Will have to update the dependency components to subscribe to these changes then if they're not already.